lib/vagrant-service-manager/action/setup_network.rb

```diff
@@ -0,0 +1,37 @@
+module Vagrant
+  module ServiceManager
+    module Action
+      class SetupNetwork
+
+        def initialize(app, env)
+          @app = app
+          @machine = env[:machine]
+          @ui = env[:ui]
+        end
+
+        def call(env)
+          add_prerequisite_network if virtualbox? && !private_network_exists?
+          @app.call(env)
+        end
+
+        private
+
+        def virtualbox?
+          @machine.provider.instance_of?(VagrantPlugins::ProviderVirtualBox::Provider)
+        end
+
+        def private_network_exists?
+          @machine.config.vm.networks.any? { |type, _| type == :private_network }
+        end
+
+        def add_prerequisite_network
+          return unless virtualbox? && !private_network_exists?
+
+          @ui.info 'Virtualbox requires an additional private network. Adding it...'
```

Comment:
@hferentschik DHCP would be a welcome step. As users run multiple instance of the Vagrantbox (usually two) and Vagrant complains about it as the IP can not be given to two vagrantbox. So if service-manager can configure the private networking with dhcp then adb-utils scripts can pick the IP address and use it for deploying OpenShift.